packages/app/src/cli/services/dev.ts

```diff
@@ -158,11 +158,13 @@ async function dev(options: DevOptions) {
   const additionalProcesses: OutputProcess[] = []
 
   if (localApp.extensions.theme.length > 0) {
-    const adminSession = await ensureAuthenticatedAdmin(storeFqdn)
-    const storefrontToken = await ensureAuthenticatedStorefront()
     const extension = localApp.extensions.theme[0]!
-    const args = await themeExtensionArgs(extension, apiKey, token, options)
-    const devExt = await devThemeExtensionTarget(args, adminSession, storefrontToken, token)
+    const [adminSession, storefrontToken, args] = await Promise.all([
+      ensureAuthenticatedAdmin(storeFqdn),
+      ensureAuthenticatedStorefront(),
+      themeExtensionArgs(extension, apiKey, token, options),
+    ])
```

Comment:
I don't think this is safe, if you run `ensureAuthenticatedAdmin` and `ensureAuthenticatedStorefront` at the same time, and the session is expired, it might try to trigger 2 authentication processes.

Reply:
yep, that's possible, we can have a convenience function to login into all 3 APIs at the same time.
Although the storefront token is only needed if there are theme extensions, but if we request them in parallel it should be OK